deploy/microshift-bootc/config-svc/api.py

```
@@ -0,0 +1,869 @@
+            image = data.get('image', '').strip()
+            
+            if not image:
+                return jsonify({'success': False, 'error': 'Image reference is required'}), 400
+            
+            # Validate image format (basic check)
+            if not (image.startswith('quay.io/') or image.startswith('docker.io/') or 
+                    ':' in image or '/' in image):
+                return jsonify({'success': False, 'error': 'Invalid image reference format'}), 400
+            
+            # Run bootc switch (this may take a while)
+            result = subprocess.run(
```

Ruff (0.14.8)

`subprocess` call: check for execution of untrusted input

(S603)